lib/angular/mediator-service.js

```diff
@@ -0,0 +1,85 @@
+var CONSTANTS = require('./constants');
+var MediatorTopicUtility = require('fh-wfm-mediator/lib/topics');
+var q = require('q');
+
+/**
+ *
+ * Getting Promises for done and error topics.
+ * This will resolve or reject the reuturned promise depending on the topic published.
+ *
+ * @param doneTopicPromise  - A promise for the done topic.
+ * @param errorTopicPromise - A promise for the error topic.
+ * @returns {Promise}
+ */
+function getTopicPromises(doneTopicPromise, errorTopicPromise) {
```

Comment:
I think we will move this to the mediator module, as it is a bit repetitive in the other PR's also.

It should DRY it up a bit.